org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/AssociateGrammarCodeLensExtensionsTest.java

```diff
@@ -144,4 +149,39 @@ public void referencedGrammarInXSDSchema() throws BadLocationException {
 				Collections.singletonList(CodeLensKind.OpenUri), //
 				cl(r(0, 0, 0, 0), "http://www.w3.org/2001/XMLSchema (with embedded xml.xsd)", OPEN_URI));
 	}
+
+	@Test
+	public void referencedGrammarUsingFileAssociation() throws BadLocationException, URISyntaxException {
+		Consumer<XMLLanguageService> configuration = ls -> {
+			ContentModelManager contentModelManager = ls.getComponent(ContentModelManager.class);
+			// Use root URI which ends with slash
+			contentModelManager.setRootURI("src/test/resources/xsd/");
+			contentModelManager.setFileAssociations(createXSDAssociationsNoNamespaceSchemaLocationLike(""));
+		};
+
+		// Use resources.xsd which defines resources as root element and @variant as
+		// required attribute
+		String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\r\n" + //
+				"  <Bad-Root></Bad-Root>";
+
+		String schemaName = AssociateGrammarCodeLensExtensionsTest.class.getResource("/xsd/resources.xsd") //

```

Comment:
Replace with

`String schemaName = XMLEntityManager.expandSystemId("", "src/test/resources/xsd/resources.xsd", true);`

and updat eexception:

`public void referencedGrammarUsingFileAssociation() throws BadLocationException, URISyntaxException, MalformedURIException {`

Reply:
For the time being I will just remove this code, since the schema name isn't used